emulator.h

```diff
@@ -83,3 +83,8 @@ Helper functions to print instruction and cpu state
 void print_instruction(uint8_t opcode);
 void print_state(i8080 *cpu);
 void print_flags(uint8_t flags);
+
+/*
+Interrupt functions
+*/
+int handle_interrupt(i8080 *cpu, uint8_t rst_instruction);
```

Comment:
we are only every passing the value to this for the 0x8 and 0x10 memory addresses correct?

Reply:
Yes but we don't pass in the actual address, we pass in which RST subroutine we want. So RST 1 maps to 0x0008 and RST 2 maps to 0x0010. So the call would be handle_interrupt(&cpu, 1) to call the subroutine starting at 0x0008.